pillar/apps/odlvideo.sls

```diff
@@ -0,0 +1,154 @@
+{% set pg_creds = salt.vault.read('postgres-{env}-odlvideo/creds/odlvideo'.format(env=ENVIRONMENT) %}
+{% set youtube_creds = salt.vault.read('secret-odl-video/global/youtube-credentials') %}
+{% set app_cert = salt.vault.read('secret-odl-video/creds/mit-application-certificate') %}
+{% set cloudfront_key = salt.vault.read('secret-odl-video/global/cloudfront-private-key') %}
+
+{% set env_dict = {
+    'ci': {
+      'domain': 'video-ci.odl.mit.edu',
+      'log_level': 'DEBUG',
+      'use_shibboleth': False,
+      'ga_id': 'UA-108097284-1',
+      'transcode_pipeline_id': '1506027488410-93oya5',
+      'youtube_project_id': 'ovs-youtube-qa'
+      },
+    'rc-apps': {
+      'domain': 'video-rc.odl.mit.edu',
+      'log_level': 'INFO',
+      'use_shibboleth': True,
+      'ga_id': 'UA-5145472-27',
+      'transcode_pipeline_id': '1506081628031-bepkel',
+      'youtube_project_id': 'ovs-youtube-qa'
+      },
+    'production-apps': {
+      'domain': 'video.odl.mit.edu',
+      'log_level': 'WARN',
+      'use_shibboleth': True,
+      'ga_id': 'UA-5145472-27',
+      'transcode_pipeline_id': '1497541042228-8mpenl',
+      'youtube_project_id': 'ovs-youtube-production'
+      }
+} %}
+
+{% set env_data = env_dict[ENVIRONMENT] %}
+
+python:
+  versions:
+    - number: 3.6.3
+      default: True
+      user: root
+
+django:
+  pip_path: {{ python_bin_dir }}/pip3
+  django_admin_path: {{ python_bin_dir }}/django-admin
+  app_name: {{ app_name }}
+  settings_module: odl_video.settings
+  automatic_migrations: True
+  app_source:
+    type: git # Options are: git, hg, archive
+    revision: release
+    repository_url: https://github.com/mitodl/odl-video-service
+    state_params:
+      - branch: release
+      - force_fetch: True
+      - force_checkout: True
+      - force_reset: True
+  environment:
+    AWS_ACCESS_KEY_ID: {{ aws_creds.data.access_key }}
+    AWS_REGION: us-east-1
+    AWS_S3_DOMAIN: s3.amazonaws.com
+    AWS_SECRET_ACCESS_KEY: {{ aws_creds.data.secret_key }}
+    CLOUDFRONT_KEY_ID: {{ cloudfront_key.data.id }}
+    CLOUDFRONT_PRIVATE_KEY: {{ cloudfront_key.data.value }}
+    DATABASE_URL: postgres://{{ pg_creds.data.username }}:{{ pg_creds.data.password }}@postgres-odlvideo.service.consul:5432/odlvideo
+    DJANGO_LOG_LEVEL: {{ env_data.log_level }}
+    DROPBOX_KEY: {{ salt.vault.read('secret-odl-video/{env}/dropbox-key'.format(env=ENVIRONMENT)).data.value }}
+    ENABLE_VIDEO_PERMISSIONS: False
+    ET_PIPELINE_ID: {{ env_data.transcode_pipeline_id }}
+    GA_DIMENSION_CAMERA: dimension1
+    GA_TRACKING_ID: {{ env_data.ga_id }}
+    LECTURE_CAPTURE_USER: {{ salt.sdb.get('consul://ovs') }}
+    MAILGUN_KEY: {{ salt.vault.read('secret-operations/global/mailgun-api-key').data.value }}
+    MAILGUN_URL: https://api.mailgun.net/v3/video.odl.mit.edu
+    MIT_WS_CERTIFICATE: {{ app_cert.data.certificate }}
+    MIT_WS_PRIVATE_KEY: {{ app_cert.data.key }}
+    ODL_VIDEO_ADMIN_EMAIL: cuddle_bunnies@mit.edu
```

Comment:
Shouldn't some of those internal email addresses (too late now probably) be kept in vault or somewhere private?